backend_typescript/server/gameApi.ts

```diff
@@ -0,0 +1,326 @@
+    const result = await game.acceptGame(user.id);
+    if (result instanceof Failure) {
+        return res.status(result.status).json({msg: result.msg});
+    }
+    return res.status(200).json({msg: 'game accepted by opponent'});
+});
+
+
+// delete a game
+gameRouter.delete('/games/:game_id', async (req, res: Response) => {
+    const user = req.user as User;
+    const game = req.game as Game;
+    const result = await game.deleteGame(user.id);
+    if (result instanceof Failure) {
+        return res.status(result.status).json({msg: result.msg});
+    }
+    return res.status(200).json({msg: 'game deleted by opponent'});
+});
+
+
+// Placing ships
+gameRouter.post('/games/:game_id/place', async (req, res: Response) => {
+    const { ship_size, start_row, start_col, end_row, end_col }: ShipBody = req.body;
+    const user = req.user as User;
+    const shipManager = req.shipManager as ShipManager;
+    const result = await shipManager.placeShip(user.id, ship_size, start_row, end_row, start_col, end_col);
+    if (result instanceof Failure) {
+        return res.status(result.status).json({msg: result.msg});
+    }
+    const finalResult = result as Success<string>;
+    return res.status(200).json({msg: finalResult.result});
+});
+
+
+// Unplace a ship
+gameRouter.delete('/games/:game_id/place', async (req, res: Response) => {
+    const { ship_size, start_row, start_col, end_row, end_col }: ShipBody = req.body;
+    const user = req.user as User;
+    const shipManager = req.shipManager as ShipManager;
+    const result = await shipManager.unplaceShip(user.id, ship_size, start_row, end_row, start_col, end_col);
+    if (result instanceof Failure) {
+        return res.status(result.status).json({msg: result.msg});
+    }
+    const finalResult = result as Success<string>;
+    return res.status(200).json({msg: finalResult.result});
+});
+
+
+// Game state change to ready / turn
+gameRouter.post('/games/:game_id/ready', async (req, res: Response) => {
+    const user = req.user as User;
+    const game = req.game as Game;
+    const result = await game.userIsReady(user.id);
+    if (result instanceof Failure) {
+        return res.status(result.status).json({msg: result.msg});
+    }
+    return res.status(200).json({msg: 'game state updated'});
+});
+
+
+// Player performs a shot
+gameRouter.post('/games/:game_id/shoot', async (req, res: Response) => {
+    const { row, col }: ShotBody = req.body;
+    const user = req.user as User;
+    const game = req.game as Game;
+    const shotResult = await game.userShoot(user.id, row, col);
+    if (shotResult instanceof Failure) {
+        return res.status(shotResult.status).json({msg: shotResult.msg});
+    }
+    const shot = shotResult as Success<string>;
+    return res.status(200).json({msg: shot.result});
+});
+
+
+// Chat api
+gameRouter.post('/games/:game_id/chat', async (req, res: Response) => {
+    const { message }: ChatMsg = req.body;
+    if (!message) {
```

Comment:
it's better to move this verification inside the function  `chatManager.postNewGameMsg`